### ISSUE TITLE
Thanks and is there a plan to release the training code?

### ISSUE THREAD

**Issue by jucic** [intel-isl/DPT#3]

**Comment by lxtGH** [comment]
I have the same question. Will you release the training code for reference?

**Comment by ranftlr** [comment]
We plan to release the training code. I can't give an exact timeline at the moment, but I hope that we'll be able to do this within one or two months.

**Comment by angrysword** [comment]
Same here. wish release training code. I used the pretrained the model (Midas) to predict depth. It is very impressive. However, one problem I am experiencing when object is a little big far away(2 or 3 meter away). It can tell the difference. I attach 2 pics for your reference
![im2](https://user-images.githubusercontent.com/3065257/117584979-df380080-b0c4-11eb-8698-f6d4d1256e67.jpg)
![im3](https://user-images.githubusercontent.com/3065257/117584982-df380080-b0c4-11eb-9469-4bb0d979b6c0.jpg)

**Comment by Tord-Zhang** [comment]
@ranftlr Wonderful work! Also looking forward to the training code. I'd like to ask how many gpus did you use for training and how long did it take to train the model?

**Comment by ranftlr** [comment]
@Tord-Zhang: We typically train on 4 Quadro 6000 cards that have 24 GB memory each. A complete run to produce the final model takes about 5 days to complete.

@angrysword: Sorry, I don't understand your question. Can you elaborate more on the problem that you are observing?

**Comment by Tord-Zhang** [comment]
@ranftlr Hi, thanks for your quick response. I am a little surprised about the training speed, since the MGDA training algorithm is used, in which a minibatch from each dataset need two forward calculation in each iteration, and the dataset is also very big. Would  Quadro 6000 be faster than Tesla V100? BTW, could I ask when the training code would be released? Thanks!

**Comment by ranftlr** [comment]
We don't go through all the images in every "epoch". Since the sizes of individual datasets can differ by an order of magnitude, we use a resampling strategy that assembles mini-batches in equal parts (on average) from every datasets. This also plays well with the diversity of the individual datasets: the large datasets typically have a lot of similar frames as the frames come from videos, whereas the smaller datasets tend to have a lot of uncorrelated images. Based on this perform a fixed number of total steps. We define an "epoch" as seeing 72000 samples and train for 2x 60 epochs - once for pre-training, once for the run over the full dataset. Please have a look at the MiDaS paper for more details (https://arxiv.org/abs/1907.01341). 

I don't have a comparison to a V100, as I don't have any available. 

There is still not exact ETA for training code.

**Comment by Tord-Zhang** [comment]
@ranftlr Did you mean sampling equal numer of images from different dataset in each batch? or sample different number of images but the same percentage? About the training, did you use DP or DDP when training with MGDA algorithm? I am not sure does MGDA support DDP.

**Comment by Tord-Zhang** [comment]
@ranftlr And which version of Blended MVS did you use? BlendedMVS, BlendedMVS+ or BlendedMVS++? High resolution or low resolution? I found that there are some unpleasant noise point in the groundtruth of low resolution blendedmvs, how did you deal with those points? Thanks.

**Comment by eliabruni** [comment]
Hello @ranftlr , still no ETA for the training code? We're considering writing our own, but that'd be undesirable especially if you are planning to release your original code.

**Comment by soroushseifi** [comment]
Hi @ranftlr, thanks for your impressive work, I would also like to mention that we at KU Leuven's PSI lab are looking forward to the training work, otherwise we will need to write our own code too, so if you could give an indication of the time you'd release the code, we will have a better idea how to organize our work.

**Comment by Tord-Zhang** [comment]
@ranftlr Hi, still no ETA for training code?

**Comment by yassineAlouini** [comment]
Did someone manage to do transfer learning using the DPT large or hybride pretrained models and using another depth dataset?
Is the loss described in this [paper](https://papers.nips.cc/paper/2014/file/7bccfde7714a1ebadf06c5f4cea752c1-Paper.pdf) a good one to use for this transfer learning?

![vns_transfer_learning_loss](https://user-images.githubusercontent.com/8986994/147224121-856aac0c-ceba-4b8c-9f68-1389b6ec3731.png)


Is unfreezing the last layer enough or should we unfreeze more?

Any pointers/tips are welcome, thanks in advance. :slightly_smiling_face:

**Comment by antocad** [comment]
Hi, we have re-implemented the paper presented in this repository, and we have added a training script. Check it out here: https://github.com/antocad/FocusOnDepth

**Comment by yassineAlouini** [comment]
@antocad Thanks for this. :ok_hand:

**Comment by Tord-Zhang** [comment]
So no plan for releasing training code?
